keymanapp/keyman
# feat(android/engine): Improve globe key experience

Addresses the immediate changes of #1887 

Description updated due to review comments...

## KeymanWeb changes (none)

## Keyman for Android changes
* Change shortpress globe actions from `GLOBE_KEY_ACTION_SHOW_MENU` to `GLOBE_KEY_ACTION_SWITCH_TO_NEXT_KEYBOARD`
* Special  lock-screen handling of system keyboard preserved
* For keyboard.html, I also found `KeymanWeb.registerStub()` was handling the package namespacing for `keyman.setActiveKeyboard()`

Remaining items on the issue (external keyboard changes, flick, and menu) are deferred to future PRs

## User Testing
@keymanapp/testers 
Setup
* Install the PR build
* Install a second Keyman keyboard
* From the Keyman Settings menu, set Keyman as a system keyboard and set as default
* From the device Settings menu, assign a lock screen password "1234" (also strongly recommend a finger print unlock if using a physical device)
* For the following tests
    * short press globe is held less than 500ms
    * long press globe is longer than 500ms
    * When verifying a selected keyboard appears, also verify the suggestion banner shows if there's an associated lexical model

## In-app keyboard  globe action
### Short press
1. In the Keyman app, short press the globe several times
2. Verify the in-app keyboard switches to the next Keyman keyboard each  time

### Long press
1. In the Keyman app, long press the globe
2. Verify the Keyboard Picker menu appears
3. Select another Keyman keyboard
4. Verify the selected keyboard appears

## System keyboard globe action
### Short press
1. Load a separate  app like Chrome and select a text field to bring up the  Keyman keyboard
2. Short press the globe several times
3. Verify the system keyboard switches to the next Keyman keyboard each time

### Long press 
1. Long press the globe 
2. Verify the Keyboard Picker menu appears
3. Select another Keyman keyboard
4. Verify the selected keyboard appears
5. Long press the globe
6. Select the "Next Input Method" button
7. Verify the next system IME is selected
8. From the system Settings, select Keyman as the system IME

### Lock the screen
1. Lock the device screen
2. For unlocking the screen, verify Keyman keyboard appears.
3. Short press the globe key
4. Verify the next Keyman keyboard appears
5. Long press the globe key
6. Verify the next Keyman keyboard appears (instead of the Keyboard Picker menu)
7. Short press the globe key several times
8. Verify eventually the next system IME appears
9. Use the system IME to unlock the device
